### PR TITLE
Add example of service_endpoints configuration

### LINKED ISSUE #141: azure_rm_subnet module generating error when using the service_endpoints parameter
##### SUMMARY
Running this as a basic test using this parameter results in the error below. 

`TypeError: string indices must be integers`

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_rm_subnet module

##### ANSIBLE VERSION

```paste below
ansible 2.9.9
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.9 (default, Apr 18 2020, 01:56:04) [GCC 8.4.0]
```

##### CONFIGURATION

```paste below
DEFAULT_VAULT_PASSWORD_FILE(env: ANSIBLE_VAULT_PASSWORD_FILE) = /home/vagrant/.ansible/.vault_password
HOST_KEY_CHECKING(env: ANSIBLE_HOST_KEY_CHECKING) = False
```

##### OS / ENVIRONMENT

Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic

##### STEPS TO REPRODUCE

main.yml
```
- hosts: azure
  collections:
    - azure.azcollection
  tasks:
  - name: Subnet
    azure_rm_subnet:
      resource_group: "dmo"
      virtual_network_name: "dmo"
      name: "frontend"
      address_prefix_cidr: "10.100.0.0/24"
      service_endpoints:
        Microsoft.Sql
    register: subnet_status
```
inventory.yml: 
```
all:
  hosts:
    azure:
      ansible_host: 127.0.0.1
      ansible_connection: local
      ansible_python_interpreter: /usr/bin/python3
    aws:
      ansible_host: 127.0.0.1
      ansible_connection: local
      ansible_python_interpreter: /usr/bin/python3
```
command: 
```
ansible-playbook -i ~/deploy/inventory.yml ~/deploy/submain.yml -vvvv
```

##### EXPECTED RESULTS
a subnet being deployed with Service Endpoints added. 

##### ACTUAL RESULTS

```paste below
TASK [Subnet] **************************************************************************************************************************************************************************************************
task path: /home/vagrant/deploy/submain.yml:5
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: vagrant
<127.0.0.1> EXEC /bin/sh -c 'echo ~vagrant && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/vagrant/.ansible/tmp `"&& mkdir /home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336 && echo ansible-tmp-1591111818.8826323-4901-96458903446336="` echo /home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336 `" ) && sleep 0'
Using module file /usr/local/lib/python3.6/dist-packages/ansible/modules/cloud/azure/azure_rm_subnet.py
<127.0.0.1> PUT /home/vagrant/.ansible/tmp/ansible-local-4829o3orv5c0/tmpeh2k3chf TO /home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/AnsiballZ_azure_rm_subnet.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/ /home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/AnsiballZ_azure_rm_subnet.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python3 /home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/AnsiballZ_azure_rm_subnet.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/AnsiballZ_azure_rm_subnet.py", line 102, in <module>
    _ansiballz_main()
  File "/home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/AnsiballZ_azure_rm_subnet.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/AnsiballZ_azure_rm_subnet.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible.modules.cloud.azure.azure_rm_subnet', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib/python3.6/runpy.py", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/modules/cloud/azure/azure_rm_subnet.py", line 399, in <module>
  File "/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/modules/cloud/azure/azure_rm_subnet.py", line 395, in main
  File "/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/modules/cloud/azure/azure_rm_subnet.py", line 235, in __init__
  File "/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/module_utils/azure_rm_common.py", line 348, in __init__
  File "/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/modules/cloud/azure/azure_rm_subnet.py", line 291, in exec_module
TypeError: string indices must be integers
fatal: [azure]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/AnsiballZ_azure_rm_subnet.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/AnsiballZ_azure_rm_subnet.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/vagrant/.ansible/tmp/ansible-tmp-1591111818.8826323-4901-96458903446336/AnsiballZ_azure_rm_subnet.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.cloud.azure.azure_rm_subnet', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/modules/cloud/azure/azure_rm_subnet.py\", line 399, in <module>\n  File \"/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/modules/cloud/azure/azure_rm_subnet.py\", line 395, in main\n  File \"/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/modules/cloud/azure/azure_rm_subnet.py\", line 235, in __init__\n  File \"/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/module_utils/azure_rm_common.py\", line 348, in __init__\n  File \"/tmp/ansible_azure_rm_subnet_payload_79babr37/ansible_azure_rm_subnet_payload.zip/ansible/modules/cloud/azure/azure_rm_subnet.py\", line 291, in exec_module\nTypeError: string indices must be integers\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

### PR DESCRIPTION
**Pull request by maartendekeizer**
##### SUMMARY
Fixes #141

Add missing example to documentation for service_endpoints.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
azure_rm_subnet

##### ADDITIONAL INFORMATION
n/a

### CODE REVIEW

**Comment by Fred-sun** [comment]
@maartendekeizer Thank you for submitting this PR. I don't think it is necessary to add this case, because tests/integration/targets/azure_rm_subnet/tasks/main.yml has already added related cases, thank you!

**Comment by maartendekeizer** [comment]
@Fred-sun Thanks for your reply. As engineer I don't want to dive-in to test files of module I use to see examples on how I can use a module. I ran in the same situation as the ts of #141 and was happy that I found that issue. Also the locations options was not very clear the first time. That is why I think we need this in the documentation.

**Comment by Fred-sun** [comment]
@maartendekeizer We need to discuss whether we need to do this, because if we put all the required Cases in the document, the module files will become a lot. Thank you!

File: plugins/modules/azure_rm_subnet.py [hunk 1]
```diff
@@ -176,6 +176,17 @@
...
+      azure_rm_subnet:
+        resource_group: myResourceGroup
+        virtual_network_name: myVirtualNetwork
+        name: mySubnet
+        address_prefix_cidr: "10.1.0.0/16"
+        service_endpoints:
+          - service: "Microsoft.Sql"
+            locations:
+              - "eastus"
+
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_subnet.py]: blank line contains whitespace
```suggestion

```

**Pull request merged**